FNPlugin/Collectors/SolarWindCollector.cs

```diff
@@ -141,41 +156,63 @@ public override void OnFixedUpdate()
 
+            return star;
+        }
 
         // calculates solar wind concentration
-        private double CalculateSolarWindConcentration()
+        private static double CalculateSolarWindConcentration(double flux)
         {
-            // check if the vessel is not in shade
-            if (!PluginHelper.lineOfSightToSun(vessel))
-            {
-                strCollectingStatus = "No solar wind detected";
-                return dConcentrationSolarWind = 0.0;
-            }
+            double dAvgKerbinSolarFlux = 1409.285; // this seems to be the average flux at Kerbin just above the atmosphere (from my tests)
+            double dAvgSolarWindPerCubM = 6000.0; // various sources differ, most state that there are around 6 particles per cm^3, so around 6000 per m^3 (some sources go up to 10/cm^3 or even down to 2/cm^3, most are around 6/cm^3).
 
-            dConcentrationSolarWind = dKerbinDistance / CalculateDistanceToSun(); // In Kerbin vicinity, the solar wind concentration will be around 100% and it will rise the closer the vessel gets to the sun
-            return dConcentrationSolarWind;
+            double dConcentration = (flux / dAvgKerbinSolarFlux) * dAvgSolarWindPerCubM;
+            return dConcentration;
         }
 
         // calculates the distance to sun
-        private double CalculateDistanceToSun()
+        private static double CalculateDistanceToSun(Vector3d vesselPosition, Vector3d sunPosition)
         {
-            Vector3d sunPosition = FlightGlobals.fetch.bodies[0].position;
-            Vector3d vesselPosition = this.part.transform.position;
-            return dDistanceFromStar = Vector3d.Distance(vesselPosition,sunPosition);
+            double dDistance = Vector3d.Distance(vesselPosition,sunPosition);
+            return dDistance;
+        }
+
+        // helper function for readying the distance for the GUI
+        private string UpdateDistanceInGUI()
+        {
+            string distance = ((CalculateDistanceToSun(part.transform.position, localStar.transform.position) - localStar.Radius) / 1000).ToString("F2") + " km";
+            return distance;
         }
 
         // the main collecting function
         private void CollectSolarWind(double deltaTimeInSeconds, bool offlineCollecting)
         {
-            dConcentrationSolarWind = CalculateSolarWindConcentration();
+            dConcentrationSolarWind = CalculateSolarWindConcentration(part.vessel.solarFlux);
 
             string strSolarWindResourceName = InterstellarResourcesConfiguration.Instance.SolarWind;
             double dPowerRequirementsMW = (double)PluginHelper.PowerConsumptionMultiplier * mwRequirements; // change the mwRequirements number in part config to change the power consumption
 
-            // check for free space in solar wind 'tanks'
+            // checks for free space in solar wind 'tanks'
             dSolarWindSpareCapacity = part.GetResourceSpareCapacity(strSolarWindResourceName);
 
```

Comment:
I don't think it should - both Hydrogen and Helium are calculated for one unit of said resource, which contains quite a lot of atoms (density of lqd Hydrogen in the real world is, according to wiki, about 0.07 g/cm3. One gram of Hydrogen = one mole of Hydrogen = so around **0.07 * 6.022 * 10^23 atoms** per cm3), whereas solar wind has only (about) **6 atoms** per that same cm3. Therefore it should be much, much less dense (around 9,96 * 10^-27 as I calculated above in KSP units, we could even fuzz it to 1 * 10^-26). If we made the resource denser than that, we might also imbalance the resource gathering a bit. Currently it is a bit too fast (with part effectivity of 100 %), but lowering the density to a realistic number should also lower the resource gains from solar wind collecting. I would like for it to be useful to players, but it shouldn't feel cheat-y :)